### ISSUE TITLE
Cannot create spark connection when master="yarn-cluster"

### ISSUE THREAD

**Issue by kevinlove** [rstudio/sparklyr#2157]
Cannot create spark connection when master="yarn-cluster":

```r
spark_version <- "2.1.0"
sc <- spark_connect(master = "yarn-cluster")

Warning message in value[[3L]](cond):
"Failed to open /ws/v1/cluster/info. Error in curl::curl_fetch_memory(url, handle = handle): URL using bad/illegal format or missing URL
"Warning message in value[[3L]](cond):
"Failed to open /ws/v1/cluster/info. Error in curl::curl_fetch_memory(url, handle = handle): URL using bad/illegal format or missing URL
"Warning message in value[[3L]](cond):
"Failed to open node1.example.com:8088/ws/v1/cluster/info. Error in if (is_http) {: argument is of length zero
"Warning message in value[[3L]](cond):
"Failed to open node2.example.com:8088/ws/v1/cluster/info. Error in if (is_http) {: argument is of length zero
"Warning message in value[[3L]](cond):
"Failed to open node1.example.com:8033/ws/v1/cluster/info. Error in if (is_http) {: argument is of length zero
"Warning message in value[[3L]](cond):
"Failed to open node2.example.com:8033/ws/v1/cluster/info. Error in if (is_http) {: argument is of length zero
"
```

Important yarn-site.xml bits:

```xml
  <property>
    <name>yarn.resourcemanager.ha.enabled</name>
    <value>true</value>
  </property>
  <property>
    <name>yarn.resourcemanager.ha.rm-ids</name>
    <value>rm1,rm2</value>
  </property>
  <property>
    <name>yarn.resourcemanager.webapp.address.rm1</name>
    <value>node1.example.com:8088</value>
  </property>
  <property>
    <name>yarn.resourcemanager.admin.address.rm1</name>
    <value>node1.example.com:8033</value>
  </property>
  <property>
    <name>yarn.resourcemanager.webapp.address.rm2</name>
    <value>node2.example.com:8088</value>
  </property>
  <property>
    <name>yarn.resourcemanager.admin.address.rm2</name>
    <value>node2.example.com:8033</value>
  </property>
```

Digging into the code, I can connect with this patch to yarn_cluster.R:

```r
spark_yarn_cluster_resource_manager_is_online <- function(rm_webapp) {
  rmQuery <- paste0(
    rm_webapp,
    "/ws/v1/cluster/info"
  )

  tryCatch({
    rmResult <- httr::GET(paste0(spark_yarn_cluster_get_protocol(), "://", rmQuery))
    if (httr::http_error(rmResult)) {
      warning("Failed to open ", rmQuery, " with status ", httr::status_code(rmResult), ". ")
      FALSE
    } else {
      TRUE
    }
  }, error = function(err) {
    warning("Failed to open ", rmQuery, ". ", err)
    FALSE
  })
}
```

**Comment by javierluraschi** [comment]
Thank you so much for identifying the fix, which is the correct one in my opining. Would be great if you could validate the fix works for you by pulling `1.0.4.9003`:

```
remotes::install_github("rstudio/sparklyr", ref = "bugfix/yarn-cluster-2157")
```

Will merge PR after tests succeed.

**Comment by kevinlove** [comment]
Looks good here, thanks for the quick turnaround!
